src/common/strtol.cc

```diff
@@ -148,41 +148,47 @@ T strict_iec_cast(std::string_view str, std::string *err)
     unit = str.substr(u, str.length() - u);
     // we accept both old si prefixes as well as the proper iec prefixes
     // i.e. K, M, ... and Ki, Mi, ...
-    if (unit.back() == 'i') {
-      if (unit.front() == 'B') {
-        *err = "strict_iecstrtoll: illegal prefix \"Bi\"";
-        return 0;
-      }
-    }
     if (unit.length() > 2) {
       *err = "strict_iecstrtoll: illegal prefix (length > 2)";
       return 0;
     }
-    switch(unit.front()) {
-      case 'K':
-        m = 10;
-        break;
-      case 'M':
-        m = 20;
-        break;
-      case 'G':
-        m = 30;
-        break;
-      case 'T':
-        m = 40;
-        break;
-      case 'P':
-        m = 50;
-        break;
-      case 'E':
-        m = 60;
-        break;
-      case 'B':
-        break;
-      default:
-        *err = "strict_iecstrtoll: unit prefix not recognized";
-        return 0;
+    if ((unit.back() == 'i') || (unit.length() == 1)) {
+      if (unit.back() == 'i') {
+        if (unit.front() == 'B') {
+          *err = "strict_iecstrtoll: illegal prefix \"Bi\"";
+          return 0;
```

Comment:
We can handle the case of `i` (single char) here itself

```suggestion
	if (unit.front() == 'i') {
	  *err = "strict_iecstrtoll: illegal prefix \"i\"";
	} else if (unit.front() == 'B') {
	  *err = "strict_iecstrtoll: illegal prefix \"Bi\"";
	}
	if (!err.empty()) {
	  return 0;
	}
```

Reply:
I don't think we do need this as `Bi` is a exception from rest of the prefixes( As `B` is valid but `Bi` is not), it needs to be handled separately. And rest of the cases are anyways handled by the switch statement.

Reply:
with the current code `i` being an invalid type would be evaluated in switch and would say `strict_iecstrtoll: unit prefix not recognized`. It should speak of `i` for clarity. Please append `unit` to the statement. (at line 184)

Reply:
Appending unit to the `err` string makes sense.